src/main/java/duke/tasks/TaskList.java

```diff
@@ -0,0 +1,306 @@
+package duke.tasks;
+
+import java.util.ArrayList;
+
+import duke.exceptions.DukeIOException;
+import duke.exceptions.DukeIllegalArgumentException;
+import duke.storage.Storage;
+
+/**
+ * A list of tasks.
+ */
+public class TaskList {
+    // Error messages
+    private static final String ERROR_MESSAGE_INDEX_OUT_OF_BOUNDS =
+            "The task number is out of range. Use \"list\" to see your tasks.";
+
+    // Message Templates
+    private static final String ADD_TASK_TEMPLATE =
+            "Got it. I've added this task:%n%s%nNow you have %d %s in the list.";
+    private static final String DELETE_TASK_TEMPLATE =
+            "Noted. I've removed this task:%n%s%nNow you have %d %s in the list.";
+
+    // An ArrayList that stores the list of tasks.
+    protected final ArrayList<Task> list;
+
+    /**
+     * Enum for types of tasks.
+     * Available types: TODO, DEADLINE, EVENT
+     */
+    public enum TaskType {
+        TODO,
+        DEADLINE,
+        EVENT
+    }
+
+    // Storage to save and load tasks from disk.
+    private final Storage taskStorage;
+
+    /**
+     * Constructor for a TaskList.
+     *
+     * @param saveFileName The name of the file to save the data to. Eg: tasks.txt
+     */
+    public TaskList(String saveFileName) {
+        this.taskStorage = new Storage(saveFileName);
+        this.list = new ArrayList<>();
+        this.importData();
+    }
+
+    // Returns "task" if there is only 1 task in the list, else returns "tasks".
+    private String taskOrTasks() {
```

Comment:
'taskOrTasks' doesn't quite sound like a verb name. 
It's tricky but might there be a way to name this method as a verb?
So far the best I've come up with is 'printTaskCheckPlural' but that's not great either...

Reply:
Hmm, okay, I will try to rename it. Thank you for your feedback!